From 1156bfd0536bb7fb5e7fc0ac9a9a9483f4d909e2 Mon Sep 17 00:00:00 2001
From: Junde Yhi <lmy441900@aosc.xyz>
Date: Wed, 13 Mar 2019 17:31:39 +0800
Subject: [PATCH] CMakeLists.txt: pass -std=c++98 to the compiler

Newer compilers that choose C++11 as their default C++ standard will
complain about several coding errors in the source code. Before we
fix those errors we tell CMake to force C++98 mode so that they can
be bypassed. However, further bug fixes to the code is still needed.

Signed-off-by: Junde Yhi <lmy441900@aosc.xyz>
---
 CMakeLists.txt | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 99a3348..961068f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -42,6 +42,14 @@ if(CMAKE_COMPILER_IS_GNUCXX AND NOT WIN32)
     add_definitions(-fvisibility=hidden)
 endif()
 
+# Set C++ standard to an older version to prevent incompatibilities to C++11
+# FIXME: Code should be fixed instead
+if(CMAKE_VERSION VERSION_LESS "3.1")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=gnu++98")
+else()
+    set(CMAKE_CXX_STANDARD 98)
+endif()
+
 ########################################################################
 # Find boost
 ########################################################################
-- 
2.18.1

